src/plugins/query_enhancements/public/search/ppl_search_interceptor.ts

```diff
@@ -68,15 +69,33 @@ export class PPLSearchInterceptor extends SearchInterceptor {
         .getDatasetService()
         .getType(datasetType);
       strategy = datasetTypeConfig?.getSearchOptions?.().strategy ?? strategy;
+
+      if (
+        dataset?.timeFieldName &&
+        datasetTypeConfig?.languageOverrides?.PPL?.hideDatePicker === false
+      ) {
+        request.params = {
+          ...request.params,
+          body: {
+            ...request.params.body,
+            timeRange: this.queryService.timefilter.timefilter.getTime(),
+          },
+        };
+      }
     }
 
     return this.runSearch(request, options.abortSignal, strategy);
   }
 
   private buildQuery() {
-    const query: Query = this.queryService.queryString.getQuery();
+    const { queryString } = this.queryService;
+    const query: Query = queryString.getQuery();
     const dataset = query.dataset;
     if (!dataset || !dataset.timeFieldName) return query;
+    const datasetService = queryString.getDatasetService();
+    if (datasetService.getType(dataset.type)?.languageOverrides?.PPL?.hideDatePicker === false)
```

Comment:
It seems when `hideDatePicker === false`, `timeRange` is added to `params.body`, otherwise, the time filter is encapsulated in the PPL query on line 102. The time filter will anyway be added, is this by intention?

Reply:
yes it's by intention, if `hideDatePicker === false`, then it's not automatically inserted. This allows custom logic to insert the time filter

Reply:
I see, so when `timeRange` is added to `params.body`, then it's up to server side search strategy to decide how to use it, did I get it right?

Reply:
right